src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/TrackerObjectManager.NativeAot.cs

```diff
@@ -0,0 +1,549 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using static System.Runtime.InteropServices.ComWrappers;
+
+namespace System.Runtime.InteropServices
+{
+    // Defined in windows.ui.xaml.hosting.referencetracker.h.
+    internal enum XAML_REFERENCETRACKER_DISCONNECT
+    {
+        // Indicates the disconnect is during a suspend and a GC can be trigger.
+        XAML_REFERENCETRACKER_DISCONNECT_SUSPEND = 0x00000001
+    };
+
+    internal struct HostServices
+    {
+        internal static IntPtr s_globalHostServices = CreateHostServices();
+
+        private static unsafe IntPtr CreateHostServices()
+        {
+            IntPtr* wrapperMem = (IntPtr*)NativeMemory.Alloc((nuint)sizeof(IntPtr));
+            wrapperMem[0] = DefaultIReferenceTrackerHostVftblPtr;
+            return (IntPtr)wrapperMem;
+        }
+
+        public static void DisconnectUnusedReferenceSources(uint flags)
+        {
+            if ((((XAML_REFERENCETRACKER_DISCONNECT)flags) & XAML_REFERENCETRACKER_DISCONNECT.XAML_REFERENCETRACKER_DISCONNECT_SUSPEND) != 0)
+            {
+                RuntimeImports.RhCollect(2, InternalGCCollectionMode.Blocking | InternalGCCollectionMode.Optimized, true);
+            }
+            else
+            {
+                GC.Collect();
+            }
+        }
+
+        public static void ReleaseDisconnectedReferenceSources()
+        {
+            GC.WaitForPendingFinalizers();
+        }
+
+        public static void NotifyEndOfReferenceTrackingOnThread()
+        {
+            ComWrappers.ReleaseExternalObjectsFromCurrentThread();
+        }
+
+        // Creates a proxy object (managed object wrapper) that points to the given IUnknown.
+        // The proxy represents the following:
+        //   1. Has a managed reference pointing to the external object
+        //      and therefore forms a cycle that can be resolved by GC.
+        //   2. Forwards data binding requests.
+        //
+        // For example:
+        //
+        // Grid <---- NoCW             Grid <-------- NoCW
```

Comment:
What's "NoCW" in this context? Not sure I've seen this acronym before 😅
Maybe clarify what that means in the comment to make this easier to understand?

Reply:
"Native Object Com Wrapper", Basically a RCW in the parlance of the legacy system.